### PR TITLE
PBL-510: Refactor historical prices to a minute precision and improve caching

### PR DESCRIPTION
**Pull request by andresmechali**
https://app.clickup.com/t/20465559/PBL-510

### CODE REVIEW

**Comment by KaiserKarel** [comment]
Task linked: [PBL-510 Subsquid - Refactor historical prices](https://app.clickup.com/t/20465559/PBL-510)

File: subsquid/src/dbHelper.ts [hunk 1]
```diff
@@ -481,27 +480,80 @@ export async function getOrCreateHistoricalAssetPrice(
...
+    } else {
+      currentPrices[assetId] = assetPrice.price;
+    }
   }
-  const json: { kusama: { usd: number }; tether: { usd: number } } = await res.json();
 
-  const picaKsmSpotPrice = await getSpotPrice(ctx, "1", "4", "2");
+  // If any price is missing, get all from Coingecko and update DB
+  if (missingPrices) {
+    const endpoint = "https://api.coingecko.com/api/v3/simple/price?ids=tether%2Ckusama&vs_currencies=usd";
```
Comment (fl-y) [inline subsquid/src/dbHelper.ts:523]: probably not the best idea to hardcode this?

File: subsquid/src/processor.ts [hunk 2]
```diff
@@ -67,18 +60,18 @@ processor.addEventHandler("BondedFinance.NewOffer", processNewOfferEvent);
...
-);
+processor.addEventHandler("Vesting.VestingScheduleAdded", processVestingScheduleAddedEvent);
 
 processor.addEventHandler("Vesting.Claimed", processVestingClaimedEvent);
 
+processor.addEventHandler("*", async ctx => {
+  // console.log(ctx.event.name);
+  // console.log(ctx.event.extrinsic);
+  // console.log(ctx.event.extrinsic?.success);
+});
```
Comment (bengalmozzi) [inline subsquid/src/processor.ts:73]: Looks like this is doing nothing. It can be removed.

Comment (andresmechali) [reply subsquid/src/processor.ts]: Done ✅

File: subsquid/src/dbHelper.ts (hunk 1 above)
Comment (andresmechali) [reply subsquid/src/dbHelper.ts]: Done @fl-y ✅

**Review by fl-y — APPROVED** [review]

**Review by bengalmozzi — COMMENTED** [review]
LGTM, but I have 1 comment :)

**Review by bengalmozzi — APPROVED** [review]
👍

**Pull request merged**